## 02_memory_leaks_in_gui.dpatch by Florian Ernst <florian@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix memory leaks in GUI code as mentioned in bug#201563

@DPATCH@
diff -urNad uae-0.8.25~/src/gtkui.c uae-0.8.25/src/gtkui.c
--- uae-0.8.25~/src/gtkui.c	2005-07-02 18:24:02.000000000 +0200
+++ uae-0.8.25/src/gtkui.c	2005-11-08 01:12:34.000000000 +0100
@@ -798,7 +798,7 @@
     uae_sem_post (&gui_sem);
     write_comm_pipe_int (&from_gui_pipe, 1, 0);
     write_comm_pipe_int (&from_gui_pipe, filesel_active, 1);
-    gtk_label_set_text (GTK_LABEL (disk_text_widget[filesel_active]), strdup (s));
+    gtk_label_set_text (GTK_LABEL (disk_text_widget[filesel_active]), s);
     filesel_active = -1;
     enable_disk_buttons (1);
     gtk_widget_destroy (disk_selector);
diff -urNad uae-0.8.25~/src/xwin.c uae-0.8.25/src/xwin.c
--- uae-0.8.25~/src/xwin.c	2005-07-02 18:24:06.000000000 +0200
+++ uae-0.8.25/src/xwin.c	2005-11-08 01:13:01.000000000 +0100
@@ -758,6 +758,7 @@
     hints->window_group = mywin;
     hints->flags = WindowGroupHint;
     XSetWMHints(display, mywin, hints);
+    XFree(hints);
 
     XMapRaised (display, mywin);
     XSync (display, 0);
